test/topology/conftest.py

```diff
@@ -219,8 +219,10 @@ async def manager(request, manager_internal, record_property, mode):
             full_url = f"<a href={request.config.getoption('artifacts_dir_url')}/{dir_path_relative}>failed_test_logs</a>"
             record_property("TEST_LOGS", full_url)
 
-    await manager_internal.after_test(test_case_name, not failed)
+    cluster_str = await manager_internal.after_test(test_case_name, not failed)
     await manager_internal.stop()  # Stop client session and close driver after each test
+    if "server_broken: True" in cluster_str:
+        pytest.fail(f"test case {test_case_name} broke ScyllaClusterManager")
```

Comment:
Maybe it is worth moving this check to the manager_client and returning not just a string, but a tuple with cluster string and boolean as a state?

Reply:
tried, it works in my local reproducer, let's see hoe CI goes